cmd/catalog/main.go

```diff
@@ -70,7 +70,8 @@ var (
 	profiling = flag.Bool(
 		"profiling", false, "serve profiling data (on port 8080)")
 
-	installPlanTimeout = flag.Duration("install-plan-retry-timeout", 1*time.Minute, "time since first attempt at which plan execution errors are considered fatal")
+	installPlanTimeout  = flag.Duration("install-plan-retry-timeout", 1*time.Minute, "time since first attempt at which plan execution errors are considered fatal")
+	bundleUnpackTimeout = flag.Duration("bundle-unpack-timeout", 10*time.Minute, "The time duration after which the bundle unpack Job for an installplan will be aborted and the installplan will be Failed. 0 is considered as having no timeout.")
```

Comment:
The timeout is only configurable per InstallPlan if you create the InstallPlan yourself.

In the more normal workflow of creating a subscription and having the catalog operator create the InstallPlan for you, the user won't have the option to retroactively apply the `bundle-unpack-timeout` annotation on the InstallPlan.
By that point the unpack Job would already be created with the default 10m timeout and won't be adjusted after the annotation is applied.

So users would have no way of tweaking the default in the standard workflow without the flag.

If this were a proper spec field then I guess we could reconcile and update/recreate the job after the annotation is applied to reflect the new timeout. But imo it's not great UX to set this on the InstallPlan after it's created, rather than setting it at the start via the Subscription.

So unless we change all of that, I'd argue we keep the flag for now if it's just the annotation.

Reply:
after having a conversation with @hasbro17 and @benluddy offline, since the annotation is mostly to make our lives easier test-wise, I think it's fine to expose this. Especially since we don't yet have info on what a great default is.